### ISSUE TITLE
On hover symbol documentation pop-up is not showing up

### ISSUE THREAD

**Issue by mihaiconstantin** [REditorSupport/vscode-R#775]
I apologize in advance for the vagueness of this issue. I did try to debug this for several hours and now, on the verge of giving up, I am posting this.

All of a sudden (i.e., with `VSCode` update `1.60.0`), the on hover documentation pop-up is not working anymore. For example, hovering over `quantile(1:100)` produces:

![image](https://user-images.githubusercontent.com/20051042/132255594-12dcddb7-91d7-4db8-9e10-5b05359b4eaf.png)

Whereas until now it used to produce:

<img width="707" alt="Untitled" src="https://user-images.githubusercontent.com/20051042/132258600-93a8b069-b21b-4ec2-b938-5242cdbc0d56.png">

---

I am running the following version of `VSCode`:

```
Version: 1.60.0
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:56:36.444Z
Electron: 13.1.8
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 20.6.0
```

The extension version is `v2.2.0`, but the same behavior is also present with the manually installed extension at  1c643ad.

I removed all extensions from `VSCode`,  cleared all leftover folders I was aware of (see below), and performed a clean install with only the `R` extension enabled.

```
rm -rf ~/Library/Preferences/com.microsoft.VSCode.helper.plist
rm -rf ~/Library/Preferences/com.microsoft.VSCode.plist
rm -rf ~/Library/Caches/com.microsoft.VSCode
rm -rf ~/Library/Caches/com.microsoft.VSCode.ShipIt/
rm -rf ~/Library/Application\ Support/Code/
rm -rf ~/Library/Saved\ Application\ State/com.microsoft.VSCode.savedState/
rm -rf ~/.vscode*
```

On the clean installed, I cleared all user settings and ensured I only use the settings necessary for the `R` extension:

```json
{
    "r.rpath.mac": "/usr/local/bin/R",
    "r.rterm.mac": "/opt/homebrew/bin/radian",
    "r.session.levelOfObjectDetail": "Detailed",
    "r.sessionWatcher": true,
    "r.bracketedPaste": true,
    "r.lsp.diagnostics": false,
    "r.plot.useHttpgd": true,
    "r.plot.defaults.fullWindowMode": true,
    "r.workspaceViewer.clearPrompt": false,
    "r.workspaceViewer.removeHiddenItems": true,
    "r.rterm.option": [
        "--no-save",
        "--no-restore"
    ]
}
```

Finally, my `.Rprofile` contains:

```r
# Config for plotting.
if (interactive() && Sys.getenv("RSTUDIO") == "") {
    Sys.setenv(TERM_PROGRAM = "vscode")
    if ("httpgd" %in% .packages(all.available = TRUE)) {
        options(vsc.plot = FALSE)
        options(device = function(...) {
            httpgd::hgd(silent = TRUE)
            .vsc.browser(httpgd::hgd_url(history = FALSE), viewer = "Beside")
        })
    }
    source(file.path(Sys.getenv(if (.Platform$OS.type == "windows") "USERPROFILE" else "HOME"), ".vscode-R", "init.R"))
}

# https://github.com/Ikuyadeu/vscode-R/issues/419
options(radian.auto_match = FALSE)
```

Can you please help me understand how to get back the documentation pop-up?

**Comment by renkun-ken** [comment]
The documentation hover is provided by language server. Could you take a look at  "Output -> R Language Server" and see if the server is started properly?

Also, you might need to turn on `r.lsp.debug` in VS Code settings and see if the language server properly handles a hover request from the language client of vscode-R.

**Comment by mihaiconstantin** [comment]
@renkun-ken, you were spot on! The language server receives the `textDocument/hover` request, but while handling it fails with `internal error: Error: argument must be coercible to non-negative integer`.

This is the full picture and stack trace:

```
[2021-09-07 12:34:52.318] received:  Content-Length: 181

[2021-09-07 12:34:52.331] received payload.

[2021-09-07 12:34:52.331] handling request:  textDocument/hover

[2021-09-07 12:34:52.333] token: {
  "full_token": "quantile",
  "right_token": "ile",
  "package": null,
  "accessor": "",
  "token": "quantile"
}

[2021-09-07 12:34:52.334] hover: {
  "uri": "file:///Users/mihai/Repositories/test/test.R",
  "version": 1
}

[2021-09-07 12:34:52.335] SYMBOL_FUNCTION_CALL quantile

[2021-09-07 12:34:52.336] guess namespace: stats

[2021-09-07 12:34:52.505] internal error: Error: argument must be coercible to non-negative integer
Call: seq_len(head.end.idx)
Stack trace:
1: repr_help_files_with_topic_generic(obj, Rd2HTML)
2: repr_html.help_files_with_topic(hfile)
3: repr::repr_html(hfile)
4: workspace$get_help(token_result$token, token_result$package)
5: hover_reply(id, uri, self$workspace, document, point)
6: self$deliver(hover_reply(id, uri, self$workspace, document, point))
7: dispatch(self, id, params)

[2021-09-07 12:34:52.507] deliver:  ["Response", "Response", "Message", "R6"]

[Error - 12:34:52 PM] Request textDocument/hover failed.
  Message: Error: argument must be coercible to non-negative integer
Call: seq_len(head.end.idx)
Stack trace:
1: repr_help_files_with_topic_generic(obj, Rd2HTML)
2: repr_html.help_files_with_topic(hfile)
3: repr::repr_html(hfile)
4: workspace$get_help(token_result$token, token_result$package)
5: hover_reply(id, uri, self$workspace, document, point)
6: self$deliver(hover_reply(id, uri, self$workspace, document, point))
7: dispatch(self, id, params)

  Code: -32603 
```

**Comment by renkun-ken** [comment]
It is caused by https://github.com/REditorSupport/languageserver/issues/468 under R 4.1.1 and has been resolved by https://github.com/REditorSupport/languageserver/pull/469. Please install the latest development version of languageserver via

```r
remotes::install_github("REditorSupport/languageserver")
```

**Comment by ManuelHentschel** [comment]
> Whereas until now it used to produce: [...]

Completely unrelated, but once you get the languageserver running again, you get even nicer hovers if you install rmarkdown and pandoc

**Comment by renkun-ken** [comment]
>Completely unrelated, but once you get the languageserver running again, you get even nicer hovers if you install rmarkdown and pandoc

Exactly. This is mostly undocumented. We should document this somewhere.

**Comment by mihaiconstantin** [comment]
Oh

> It is caused by [REditorSupport/languageserver#468](https://github.com/REditorSupport/languageserver/issues/468) under R 4.1.1 and has been resolved by [REditorSupport/languageserver#469](https://github.com/REditorSupport/languageserver/pull/469). Please install the latest development version of languageserver via
> 
> ```r
> remotes::install_github("REditorSupport/languageserver")
> ```

Thank you, @renkun-ken. Indeed, installing from GitHub fixed it!

**Comment by mihaiconstantin** [comment]
> > Whereas until now it used to produce: [...]
> 
> Completely unrelated, but once you get the languageserver running again, you get even nicer hovers if you install rmarkdown and pandoc

Very handy that you mentioned it. Looks better indeed, and I don't recall seeing this mentioned in the documentation.

![image](https://user-images.githubusercontent.com/20051042/132345515-afcfa76a-f11d-4b27-ae7a-ff4849a43b5f.png)
